code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceConsumeMapDaoImpl.java

```diff
@@ -66,4 +66,15 @@ public ServiceConsumeMap findNonRemovedMap(long serviceId, long consumedServiceI
                                 .and(SERVICE_CONSUME_MAP.REMOVED.isNull())).fetchInto(ServiceConsumeMapRecord.class);
     }
 
+    @Override
+    public List<? extends ServiceConsumeMap> findConsumingMapsToRemove(long serviceId) {
+        return create()
+                .selectFrom(SERVICE_CONSUME_MAP)
+                .where(
+                        SERVICE_CONSUME_MAP.CONSUMED_SERVICE_ID.eq(serviceId)
+                                .and((SERVICE_CONSUME_MAP.REMOVED.isNull().
+                                        or(SERVICE_CONSUME_MAP.STATE.eq(CommonStatesConstants.REMOVING))))).
```

Comment:
Here I'm finding the maps To remove; so I specifically looking for maps with removed flag null, or state=Removing (records against which Remove process was called once). I follow the logic from genericMapDao here